### DISCUSSION TITLE
Creating Separate Directories for Frontend and Backend.

### DISCUSSION THREAD

**Discussion by tejasbenibagde** [ANSHIKA-26/WordWise#1098] [Ideas]
Hello @ANSHIKA-26,

I’d like to discuss the possibility of separating the frontend and backend of our project. Right now, we have a dedicated folder for the backend, which is a great start. However, the frontend files are quite scattered, making it difficult to manage.

By separating them more clearly, we could improve our code structure significantly. This would not only enhance maintainability but also simplify the process of hosting the frontend and backend on different routes. Ultimately, this approach would help us establish better API connections between the two.

I believe this change could lead to a more organized workflow and make future development smoother. What do you think?

**Comment by tejasbenibagde** [comment]
By organizing the frontend into a dedicated directory instead of placing all the files in the root of the project, we can achieve a clearer code structure. This change will facilitate the implementation of a global state management system using Redux and Redux Toolkit, which will help us maintain consistent API connections across all pages of the application.

A global state management system will also simplify authentication by allowing us to use a single hook in our JavaScript code, rather than repeating lengthy, complex code across multiple files. This approach will not only streamline our code but also enhance overall maintainability.


@ANSHIKA-26 I'm looking for your thoughts in this matter.

**Reply by ANSHIKA-26** [reply]
sounds good
